### PR TITLE
feat: Modbus Rtu over tcp enhancement

### PR DESCRIPTION
**Pull request by mbezuidenhout**
### Required for all PRs:

- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.

resolves #

Added support for RTU over TCP

### CODE REVIEW

**Comment by mbezuidenhout** [comment]
Unfortunately package that the test suite uses to make a tcp modbus server does not offer a RTU over TCP option. I have tested the changes on a live system and can confirm that it works.

`!signed-cla`

File: plugins/inputs/modbus/modbus.go [hunk 1]
```diff
@@ -88,6 +88,10 @@ const sampleConfig = `
 
   # TCP - connect via Modbus/TCP
   controller = "tcp://localhost:502"
+  
+  # RTU over TCP - connect via Modbus RTU over TCP
+  # controller = "tcp://localhost:502"
+  # transmission_mode = "RTU"
```
Comment (srebhan) [inline plugins/inputs/modbus/modbus.go:94]: I think we should either place this further down and fuse in the serial `transmission_mode` setting or make more clear that this is the setting for TCP, e.g. by moving the setting closer to the `controller` setting and place the comment above `controller`.
Please also add a comment like
```toml
  ## For Modbus over TCP you can choose between "TCP", "RTUoverTCP" and "ASCIIoverTCP".
```
and make "TCP" the default under the TCP `controller` setting.

File: plugins/inputs/modbus/modbus.go [hunk 2]
```diff
@@ -246,9 +250,16 @@ func (m *Modbus) initClient() error {
...
+		switch m.TransmissionMode {
+		case "RTU":
+			handler := mb.NewRTUOverTCPClientHandler(host + ":" + port)
+			handler.Timeout = time.Duration(m.Timeout)
+			m.handler = handler
+		default:
+			handler := mb.NewTCPClientHandler(host + ":" + port)
+			handler.Timeout = time.Duration(m.Timeout)
+			m.handler = handler
+		}
```
Comment (srebhan) [inline plugins/inputs/modbus/modbus.go:262]: Please use `"RTUoverTCP"` and also implement `"ASCIIoverTCP"` in the same PR.

**Review by srebhan — COMMENTED** [review]
Thanks @mbezuidenhout for this PR!

The code looks mostly good, my only comments are using "RTUoverTCP" to make it more obvious to the user what is going on. Furthermore, please also implement "ASCIIoverTCP" to complete the features. :-)

Additionally, please make sure that the default setting shown in the config file is actually the default behavior (i.e. "MODBUS/TCP" without "RTU" or "ASCII").

File: plugins/inputs/modbus/modbus.go (hunk 1 above)
Comment (mbezuidenhout) [reply plugins/inputs/modbus/modbus.go]: I've added a more descriptive message for the different `transmission_mode`s. Transmission mode has been condensed to only one description in the README.

File: plugins/inputs/modbus/modbus.go (hunk 2 above)
Comment (mbezuidenhout) [reply plugins/inputs/modbus/modbus.go]: The different transmission modes now are: `RTUoverTCP`, `ASCIIoverTCP`, `TCP` (default) for TCP, `RTU` and `ASCII`

**Review by srebhan — APPROVED** [review]
Perfect, thank you @mbezuidenhout!

**Review by ssoroka — APPROVED** [review]

**Pull request merged**